compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/node/SnapshotInvalidationTracker.skiko.kt

```diff
@@ -76,6 +82,21 @@ internal class SnapshotInvalidationTracker(
         Snapshot.sendApplyNotifications()
         snapshotChanges.perform()
     }
+
+    /**
+     * [ComposeScene.render] should wrap itself in this; makes sure that inside `render`, the calls
+     * to [OwnerSnapshotObserver] are performed synchronously.
+     *
+     * See [OwnerSnapshotObserverTest.observeReadsChangedBeforeDisposeEffect] for more details.
+     */
+    inline fun rendering(crossinline block: () -> Unit) {
```

Comment:
Let's not introduce naming and comment references to another class. I'd name it something like this:
```suggestion
    inline fun executeSnapshotCommandsSynchronously(crossinline block: () -> Unit) {
```

Reply:
`SnapshotInvalidationTracker` is already (conceptually) tightly coupled with `ComposeScene`. It has properties like `needLayout` and functions like `onDraw`.

Reply:
If we do this not only in `render` method (as we discuss in the other discussion), we should rename it.

Reply:
@igordmn Already done.